ipam/allocator.go

```diff
@@ -126,12 +126,14 @@ func (alloc *Allocator) cancelOps(ops *[]operation) {
 // if we found any.
 func (alloc *Allocator) cancelOpsFor(ops *[]operation, ident string) bool {
 	var found bool
-	for i, op := range *ops {
-		if op.ForContainer(ident) {
+	for i := 0; i < len(*ops); {
+		if op := (*ops)[i]; op.ForContainer(ident) {
 			found = true
 			op.Cancel()
 			*ops = append((*ops)[:i], (*ops)[i+1:]...)
+			continue
 		}
+		i++
```

Comment:
I've been reading the golang spec and cannot figure out what is going wrong with the original code. There is obviously some issue with modifying the slice while iterating over it. Presumably `i` ends up pointing beyond the end of the slice. But why? And what is the value of `op`?